### ISSUE TITLE
NullPointerException thrown when publication contains nested levels of chunking

### ISSUE THREAD

**Issue by radu-pisoi** [dita-ot/dita-ot#2843]
## Expected Behavior

The XHTML transformation should work without errors when the publication contains nested levels of chunking.

The DITA Map used to reproduce the problem has the next simplified structure:
```
main.ditamap
<mapref href="ditamap_level2.ditamap" format="ditamap" type="map"/>
---
**  ditamap_level2.ditamap
**  <map chunk="to-content">
**    <title>DITA MAp Level 2</title>       
**    <mapref href="ditamap_level3.ditamap" format="ditamap" type="map"/>
**  </map>
---
****    ditamap_level3.ditamap
****	<map chunk="to-content">
****      <title>DITA Map Level 3</title>
****      <topicref href="Topics/test.dita" type="concept"/>       
****    </map>
```

## Actual Behavior
The XHTML transformation fails with ``java.lang.NullPointerException`` in chunk processing.

```
chunk:
    [chunk] Processing file:/D:/test-oxygen/dita-ot-chunk-issue/temp/xhtml/oxygen_dita_temp/main.ditamap
    [chunk] Processing file:/D:/test-oxygen/dita-ot-chunk-issue/temp/xhtml/oxygen_dita_temp/Chunk1025069204.dita
    [chunk] Processing file:/D:/test-oxygen/dita-ot-chunk-issue/temp/xhtml/oxygen_dita_temp/Topics/test.dita
    [chunk] Writing file:/D:/test-oxygen/dita-ot-chunk-issue/temp/xhtml/oxygen_dita_temp/Chunk456609017.dita

BUILD FAILED
D:\Program Files\Oxygen XML Editor 19.1\frameworks\dita\DITA-OT2.x\build.xml:45: The following error occurred while executing this line:
D:\Program Files\Oxygen XML Editor 19.1\frameworks\dita\DITA-OT2.x\plugins\org.dita.base\build_preprocess.xml:323: java.lang.NullPointerException
	at org.dita.dost.writer.ChunkTopicParser.processChunk(ChunkTopicParser.java:239)
	at org.dita.dost.writer.ChunkTopicParser.write(ChunkTopicParser.java:61)
	at org.dita.dost.reader.ChunkMapReader.processCombineChunk(ChunkMapReader.java:542)
	at org.dita.dost.reader.ChunkMapReader.processTopicref(ChunkMapReader.java:347)
	at org.dita.dost.reader.ChunkMapReader.process(ChunkMapReader.java:143)
	at org.dita.dost.writer.AbstractDomFilter.read(AbstractDomFilter.java:55)
	at org.dita.dost.reader.ChunkMapReader.read(ChunkMapReader.java:119)
	at org.dita.dost.module.ChunkModule.execute(ChunkModule.java:80)
	at org.dita.dost.pipeline.PipelineFacade.execute(PipelineFacade.java:80)
	at org.dita.dost.invoker.ExtensibleAntInvoker.execute(ExtensibleAntInvoker.java:230)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
	at sun.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:435)
	at org.apache.tools.ant.Target.performTasks(Target.java:456)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1405)
	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:38)
```

## Steps to Reproduce
1. Run the XHTML transformation for the attached DITA Map.
[test_map.zip](https://github.com/dita-ot/dita-ot/files/1474210/test_map.zip)

## Environment

* DITA-OT version: 2.5.2
* Operating system and version:  _Windows_
* How did you run DITA-OT? oXygen
* Transformation type: XHTML

**Comment by radu-pisoi** [comment]
I have attached the test DITA Map.

[test_map.zip](https://github.com/dita-ot/dita-ot/files/1474203/test_map.zip)

**Comment by robander** [comment]
Reproduced with the sample files in 3.0, using both `xhtml` and `pdf` transform types (so it's not fixed in `preprocess2`.

**Comment by robander** [comment]
Just encountered another simpler variant of this that may show the likely exception - I think this is caused by a chunked branch that has a chunked topicgroup inside. The sub-map is converted to a topicgroup container when the map references are resolved, and `chunk="to-content"` from the submap is preserved on that group; this led me to try the following map, which also ends in a null pointer error. The two topics are very basic, just a title and paragraph:
```
<!DOCTYPE map PUBLIC "-//OASIS//DTD DITA Map//EN" "ibm-map.dtd">
<map xml:lang="en-us"><title>Feeling chunky</title>
<topicref href="container.dita" chunk="to-content">
<topicgroup chunk="to-content">
<topicref href="fromsub.dita"/>
</topicgroup>
</topicref>
</map>
```
[jira129.zip](https://github.com/dita-ot/dita-ot/files/2852402/jira129.zip)

**Comment by raducoravu** [comment]
@robander are these two other issues duplicates with this one?
https://github.com/dita-ot/dita-ot/issues/2730
https://github.com/dita-ot/dita-ot/issues/2428

**Comment by robander** [comment]
@raducoravu definitely a duplicate of #2730 but I think #2428 is different.

**Comment by robander** [comment]
Updated test case: 
[jira129.zip](https://github.com/dita-ot/dita-ot/files/2862445/jira129.zip)

Now includes:

- Container topic with `chunk="to-content"`
- nested topicgroup with `chunk="to-content"` (currently causes NPE); ideally this would create a new chunk with generated name, but not add in an extra generated title
- nested topichead with `chunk="to-content"` (also causes NPE); ideally this would create a new chunk with generated name, including the title as the root "topic"
- nested topicref with `chunk="to-content"` and no title + no href, but includes shortdesc (also causes NPE); edge case, it should not crash, should instead create new chunk with generated name, empty title but use the short description

**Comment by robander** [comment]
Fixed with #3216.
